### PR TITLE
fix: using test data sample for catalog example

### LINKED ISSUE #8041: Error from example `catalog`
### Describe the bug

For the catalog test in the `example` directory, when running the `dict-page-offset-zero.parquet` file, an unparsable exception will be thrown.

### To Reproduce

```cargo run --example catalog```

``` Parquet error: Parquet error: Could not parse metadata: protocol error```

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by korowa**
## Which issue does this PR close?

Closes #8041.

## Rationale for this change

The purpose of `catalog.rs` to provide an example of `SchemaProvider` implementation, so there is no need to use submodule which may contain non-standard / non-readable .parquet files, and .csv with some sample should be enough.

## What changes are included in this PR?

Replacing `[parquet-]testing` submodules usage in catalog example with directories and .csv containing basic sample.

## Are these changes tested?

`catalog.rs` execution now included (back) in the CI script for examples.

## Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion-examples/examples/catalog.rs [hunk 1]
```diff
@@ -36,94 +33,102 @@ use datafusion::{
...
 
 #[tokio::main]
 async fn main() -> Result<()> {
-    let repo_dir = std::fs::canonicalize(
-        PathBuf::from(env!("CARGO_MANIFEST_DIR"))
-            // parent dir of datafusion-examples = repo root
-            .join(".."),
-    )
-    .unwrap();
+    env_logger::builder()
```
Comment (alamb) [inline datafusion-examples/examples/catalog.rs:42]: 👍

File: datafusion-examples/examples/catalog.rs [hunk 2]
```diff
@@ -36,94 +33,102 @@ use datafusion::{
...
     // register our catalog in the context
     ctx.register_catalog("dircat", Arc::new(catalog));
     {
         // catalog was passed down into our custom catalog list since we overide the ctx's default
         let catalogs = catlist.catalogs.read().unwrap();
         assert!(catalogs.contains_key("dircat"));
     };
-    // take the first 5 (arbitrary amount) keys from our schema's hashmap.
+
+    // take the first 3 (arbitrary amount) keys from our schema's hashmap.
```
Comment (alamb) [inline datafusion-examples/examples/catalog.rs:91]: Thie code still seems to take 5 (not 3)

```suggestion
    // take the first 5 (arbitrary amount) keys from our schema's hashmap.
```

**Review by alamb — APPROVED** [review]
Thank you @korowa  -- this looks like a great improvement

I also tested this PR manually:

```shell
andrewlamb@Andrews-MacBook-Pro:~/Software/arrow-datafusion$ cargo run  --example catalog
    Finished dev [unoptimized + debuginfo] target(s) in 0.17s
     Running `target/debug/examples/catalog`
[2024-02-28T00:58:41Z INFO  catalog] querying table 2.csv from schema_a
[2024-02-28T00:58:41Z INFO  catalog] query completed
+-----+-------+
| key | value |
+-----+-------+
| 1   | foo   |
| 2   | bar   |
| 3   | baz   |
+-----+-------+
[2024-02-28T00:58:41Z INFO  catalog] querying table 3.csv from schema_a
[2024-02-28T00:58:41Z INFO  catalog] query completed
+-----+-------+
| key | value |
+-----+-------+
| 1   | foo   |
| 2   | bar   |
| 3   | baz   |
+-----+-------+
[2024-02-28T00:58:41Z INFO  catalog] querying table 4.csv from schema_a
[2024-02-28T00:58:41Z INFO  catalog] query completed
+-----+-------+
| key | value |
+-----+-------+
| 1   | foo   |
| 2   | bar   |
| 3   | baz   |
+-----+-------+
[2024-02-28T00:58:41Z INFO  catalog] querying table 1.csv from schema_a
[2024-02-28T00:58:41Z INFO  catalog] query completed
+-----+-------+
| key | value |
+-----+-------+
| 1   | foo   |
| 2   | bar   |
| 3   | baz   |
+-----+-------+
[2024-02-28T00:58:41Z INFO  catalog] querying table 0.csv from schema_a
[2024-02-28T00:58:41Z INFO  catalog] query completed
+-----+-------+
| key | value |
+-----+-------+
| 1   | foo   |
| 2   | bar   |
| 3   | baz   |
+-----+-------+
[2024-02-28T00:58:41Z INFO  catalog] dropping table 2.csv
```

File: datafusion-examples/examples/catalog.rs (hunk 2 above)
Comment (korowa) [reply datafusion-examples/examples/catalog.rs:91]: Thank you, fixed by actually taking 3 tables -- the intention was to slightly reduce console output while running the example.

**Comment by alamb** [comment]
Thanks again @korowa

**Pull request merged**